Move jobframework constants into the directory

mimowo opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

/kind cleanup
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:

To make the jobframework stuff contained within the package. See related comment: https://github.com/kubernetes-sigs/kueue/pull/633#discussion_r1137383502

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Part of #369

#### Special notes for your reviewer:

The solution is to extract from the monolitic `testing` package the packages for job framework testing (batchv1.Job and MPIJob). Only these packages will depend on the `jobframwork` package.

alculquicondor commented on pkg/controller/jobframework/constants.go:
```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2022 The Kubernetes Authors.
```
2023

mimowo replied:
Done

alculquicondor commented on pkg/controller/jobframework/reconciler.go:
```diff
@@ -105,7 +105,7 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 	// queue-name or the parent-workload annotation set.
 	if !r.manageJobsWithoutQueueName && job.QueueName() == "" && isStandaloneJob {
-		log.V(3).Info(fmt.Sprintf("Neither %s, nor %s annotation is set, ignoring the job", constants.QueueAnnotation, constants.ParentWorkloadAnnotation))
+		log.V(3).Info(fmt.Sprintf("Neither %s, nor %s annotation is set, ignoring the job", QueueAnnotation, ParentWorkloadAnnotation))
```
```suggestion
		log.V(3).Info(fmt.Sprintf("Neither %s label, nor %s annotation is set, ignoring the job", QueueLabel, ParentWorkloadAnnotation))
```

alculquicondor commented on pkg/controller/jobframework/validation.go:
```diff
@@ -19,14 +19,12 @@ import (
-	ParentWorkloadKeyPath = annotationsPath.Key(constants.ParentWorkloadAnnotation)
-	QueueNamePath         = annotationsPath.Key(constants.QueueAnnotation)
+	ParentWorkloadKeyPath = annotationsPath.Key(ParentWorkloadAnnotation)
+	QueueNamePath         = annotationsPath.Key(QueueAnnotation)
```
oh, we should validate the Label as well. I forgot to do this as part of the migration to v1beta1.

alculquicondor commented on pkg/controller/jobs/job/job_controller.go:
```diff
@@ -118,11 +117,11 @@ func (j *Job) Object() client.Object {
 }
 
 func (j *Job) QueueName() string {
-	return j.Annotations[constants.QueueAnnotation]
```
I think this was lost over the rebases.... We should check first in Labels, and, if not found, as Annotation.

mimowo replied:
Possibly, but if so, there wasn't a test for it, I'm going to add the support and the test, but I would like to move the function out from the interface as discussed here: https://github.com/kubernetes-sigs/kueue/pull/640#discussion_r1140250234.

alculquicondor commented on pkg/controller/jobs/mpijob/mpijob_controller.go:
```diff
@@ -64,11 +63,11 @@ func (j *MPIJob) Object() client.Object {
 
 func (j *MPIJob) QueueName() string {
-	return j.Annotations[constants.QueueAnnotation]
+	return j.Annotations[jobframework.QueueAnnotation]
```
same change here

mimowo replied:
Fixed

alculquicondor commented on pkg/util/testingjobs/job/wrappers.go:
```diff
@@ -0,0 +1,122 @@
+
+// Queue updates the queue name of the job
+func (j *JobWrapper) Queue(queue string) *JobWrapper {
+	j.Annotations[jobframework.QueueAnnotation] = queue
```
label

mimowo replied:
done

mimowo commented:
@alculquicondor I think I have addressed the comments and simplified the interface so that the handling of labels and annotations for queue name is commonized. If we think MPIJob should support the spec.runPolicy.schedulingPolicy.Queue I will add this under a dedicated PR. This didn't work anyway, so let's progress the interface as needed.

alculquicondor commented on pkg/controller/jobframework/validation.go:
```diff
@@ -19,16 +19,22 @@ import (
+	return allErrs
+}
+
 func ValidateAnnotationAsCRDName(job GenericJob, crdNameAnnotation string) field.ErrorList {
```
can be unexported now?

mimowo replied:
Still is used by parent-workload, could follow up as for Queuename

alculquicondor commented on test/integration/controller/job/job_controller_test.go:
```diff
@@ -243,6 +243,17 @@ var _ = ginkgo.Describe("Job controller", func() {
 				createdWorkload.Status.Conditions[0].Status == metav1.ConditionTrue
 		}, util.Timeout, util.Interval).Should(gomega.BeTrue())
 	})
+	ginkgo.It("Should reconcile job when queueName set by annotation (deprecated)", func() {
```
Thanks!

Add an empty line before `It`

mimowo replied:
Done

mimowo commented on test/integration/controller/job/job_controller_test.go:
```diff
@@ -243,6 +243,17 @@ var _ = ginkgo.Describe("Job controller", func() {
+		}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		gomega.Expect(createdWorkload.Spec.QueueName).Should(gomega.Equal(jobQueueName))
+	})
 	ginkgo.When("The parent-workload annotation is used", func() {
```
Done

mimowo commented on pkg/controller/jobframework/validation.go:
```diff
@@ -39,19 +45,29 @@ func ValidateAnnotationAsCRDName(job GenericJob, crdNameAnnotation string) field
 	return allErrs
 }
 
+func ValidateLabelAsCRDName(job GenericJob, crdNameLabel string) field.ErrorList {
```
Yes, lowercased
